File: Program's_Contributed_By_Contributors/Python_Programs/TicTacToe.py
# Initialize the Tic-Tac-Toe board
board = [" " for _ in range(9)]

# Function to display the Tic-Tac-Toe board
def display_board(board):
    print(f"{board[0]} | {board[1]} | {board[2]}")
    print("---------")
    print(f"{board[3]} | {board[4]} | {board[5]}")
    print("---------")
    print(f"{board[6]} | {board[7]} | {board[8]}")

# Function to check if a player has won
def check_win(board, player):
    # Check rows, columns, and diagonals
    win_conditions = [
        [0, 1, 2], [3, 4, 5], [6, 7, 8],  # Rows
        [0, 3, 6], [1, 4, 7], [2, 5, 8],  # Columns
        [0, 4, 8], [2, 4, 6]              # Diagonals
    ]
    
    for condition in win_conditions:
        if all(board[i] == player for i in condition):
            return True
    return False

# Function to check if the board is full
def is_board_full(board):
    return " " not in board

# Main game loop
current_player = "X"

while True:
    display_board(board)
    move = int(input(f"Player {current_player}, enter your move (1-9): ")) - 1

    if board[move] == " ":
        board[move] = current_player
        if check_win(board, current_player):
            display_board(board)
            print(f"Player {current_player} wins!")
            break
        elif is_board_full(board):
            display_board(board)
            print("It's a tie!")
            break
        current_player = "O" if current_player == "X" else "X"
    else:
        print("Invalid move. Try again.")

File: Program's_Contributed_By_Contributors/C++ Programs/tic tac toe game/tictactoe.c
#include <stdio.h>

// Function to initialize the Tic-Tac-Toe board
void initializeBoard(char board[3][3]) {
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            board[i][j] = ' ';
        }
    }
}

// Function to print the Tic-Tac-Toe board
void printBoard(char board[3][3]) {
    printf("-------------\n");
    for (int i = 0; i < 3; i++) {
        printf("| %c | %c | %c |\n", board[i][0], board[i][1], board[i][2]);
        printf("-------------\n");
    }
}

// Function to check if the game is over
int isGameOver(char board[3][3]) {
    // Check rows
    for (int i = 0; i < 3; i++) {
        if (board[i][0] != ' ' && board[i][0] == board[i][1] && board[i][1] == board[i][2]) {
            return 1;
        }
    }
    
    // Check columns
    for (int j = 0; j < 3; j++) {
        if (board[0][j] != ' ' && board[0][j] == board[1][j] && board[1][j] == board[2][j]) {
            return 1;
        }
    }
    
    // Check diagonals
    if (board[0][0] != ' ' && board[0][0] == board[1][1] && board[1][1] == board[2][2]) {
        return 1;
    }
    if (board[0][2] != ' ' && board[0][2] == board[1][1] && board[1][1] == board[2][0]) {
        return 1;
    }
    
    // Check for a draw
    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 3; j++) {
            if (board[i][j] == ' ') {
                return 0;
            }
        }
    }
    
    return -1; // Game is a draw
}

// Function to make a move
void makeMove(char board[3][3], int row, int col, char player) {
    if (board[row][col] == ' ') {
        board[row][col] = player;
    }
}

int main() {
    char board[3][3];
    char currentPlayer = 'X';
    int row, col;
    int gameOver = 0;
    
    initializeBoard(board);
    
    printf("Welcome to Tic-Tac-Toe!\n");
    
    while (!gameOver) {
        printf("Current board:\n");
        printBoard(board);
        
        printf("Player %c, enter your move (row and column): ", currentPlayer);
        scanf("%d %d", &row, &col);
        
        if (row >= 0 && row < 3 && col >= 0 && col < 3) {
            if (board[row][col] == ' ') {
                makeMove(board, row, col, currentPlayer);
                gameOver = isGameOver(board);
                if (gameOver == -1) {
                    printf("It's a draw!\n");
                } else if (gameOver) {
                    printf("Player %c wins!\n", currentPlayer);
                } else {
                    currentPlayer = (currentPlayer == 'X') ? 'O' : 'X';
                }
            } else {
                printf("Invalid move. That cell is already occupied.\n");
            }
        } else {
            printf("Invalid move. Please enter row and column values between 0 and 2.\n");
        }
    }
    
    printf("Final board:\n");
    printBoard(board);
    
    return 0;
}